cas/service.c

```diff
@@ -2181,7 +2184,18 @@ static int cas_exec(struct cas_fom *fom, enum m0_cas_opcode opc,
 		ret = m0_ctg_insert(ctg_op, ctg, &kbuf, &vbuf, next);
 		break;
 	case CTG_OP_COMBINE(CO_DEL, CT_BTREE):
-		ret = m0_ctg_delete(ctg_op, ctg, &kbuf, next);
+		ret = m0_ctg_lookup_delete(ctg_op, ctg, &kbuf, &lbuf, flags, next);
+		if (ctg_op->co_rc == 0) {
+			rec = cas_at(cas_op(fom0), rec_pos);
+
+			/*
+			 * Here @lbuf is allocated in m0_ctg_lookup_delete()
+			 * and released in cas_fom_fini().
+			 */
+			m0_rpc_at_init(&rec->cr_val);
+			rec->cr_val.ab_type = M0_RPC_AT_INLINE;
+			rec->cr_val.u.ab_buf = lbuf;
+		}
```

Comment:
If failure happesn, the "lbuf" is leaked?

Please refer to https://github.com/Seagate/cortx-motr/pull/966/files#r691774431